app/src/main/java/org/mozilla/fenix/home/PocketUpdatesMiddleware.kt

```diff
@@ -4,25 +4,63 @@
 
 package org.mozilla.fenix.home
 
+import androidx.datastore.core.DataStore
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 import mozilla.components.lib.state.Action
 import mozilla.components.lib.state.Middleware
 import mozilla.components.lib.state.MiddlewareContext
 import mozilla.components.service.pocket.PocketStoriesService
+import org.mozilla.fenix.datastore.SelectedPocketStoriesCategories
+import org.mozilla.fenix.datastore.SelectedPocketStoriesCategories.SelectedPocketStoriesCategory
+import org.mozilla.fenix.home.sessioncontrol.viewholders.pocket.PocketRecommendedStoriesSelectedCategory
 
 /**
  * [HomeFragmentStore] middleware reacting in response to Pocket related [Action]s.
+ *
+ * @param coroutineScope [CoroutineScope] used for long running operations like disk IO.
+ * @param pocketStoriesService [PocketStoriesService] used for updating details about the Pocket recommended stories.
+ * @param selectedPocketCategoriesStore [DataStore] used for reading or persisting details about the
+ * currently selected Pocket recommended stories categories.
  */
 class PocketUpdatesMiddleware(
     private val coroutineScope: CoroutineScope,
-    private val pocketStoriesService: PocketStoriesService
+    private val pocketStoriesService: PocketStoriesService,
+    private val selectedPocketCategoriesStore: DataStore<SelectedPocketStoriesCategories>
 ) : Middleware<HomeFragmentState, HomeFragmentAction> {
     override fun invoke(
         context: MiddlewareContext<HomeFragmentState, HomeFragmentAction>,
         next: (HomeFragmentAction) -> Unit,
         action: HomeFragmentAction
     ) {
+        // Pre process actions
+        when (action) {
+            is HomeFragmentAction.PocketStoriesCategoriesChange -> {
+                // Whenever the stories categories are updated we need to restore their "selected" status.
+
+                coroutineScope.launch {
+                    selectedPocketCategoriesStore.data.collect { persistedSelectedCategories ->
+                        // Replace the original action with one that also
```

Comment:
"that also" ...? :) (It's also no longer "replacing" I guess)

Reply:
Thinking of changing to
> // Intercept the original action which would only update categories and
// dispatch a new action which also updates which categories are selected by the user
// from previous locally persisted data.